src/robot.py

```diff
@@ -152,8 +166,9 @@ def get_all_actions(self, pose, motion_planner):
         coordinates.append((x_base, y_base, cw_turn))
 
         for x, y, theta in coordinates:
-            if self.is_cord_inbounds((x, y, theta)) and (not self.orchestrator.is_pt_locked((x, y)) or (x, y) == (self.end_pose[0], self.end_pose[1])):
+            if self.is_cord_inbounds((x, y, theta)) and (not self.orchestrator.is_pt_locked((x, y)) or (x, y) == (self.shelf_pose[0], self.shelf_pose[1])):
```

Comment:
any reason not to use goal pose here?  that way as we assign new goals, we dont need to update the shelf pose everytime as well?

Reply:
Leaving it at goal_pose introduced a bug. Basically there's a case where the robot has reached the shelf, and the new goal_pose is set to the charging station. In this case, all poses where the robot tries to execute a turn are suddenly blocked because the shelf is considered an obstacle.